src/main/java/net/earthcomputer/clientcommands/command/WhisperEncryptedCommand.java

```diff
@@ -34,73 +38,66 @@
-        Optional<PlayerListEntry> entry = profiles.stream().findFirst().flatMap(gp ->
-            source.getClient().getNetworkHandler().getPlayerList().stream().filter(e -> e.getProfile().getName().equalsIgnoreCase(gp.getName())).findFirst()
-        );
-        if (entry.isEmpty()) {
-            source.sendError(Text.translatable("commands.cwhisperencrypted.playerNotFound"));
-            return 0;
+        if (profiles.size() != 1) {
+            throw PLAYER_NOT_FOUND_EXCEPTION.create();
         }
+        PlayerListEntry recipient = source.getClient().getNetworkHandler().getPlayerList().stream()
+                .filter(p -> p.getProfile().getName().equalsIgnoreCase(profiles.iterator().next().getName()))
+                .findFirst()
+                .orElseThrow(PLAYER_NOT_FOUND_EXCEPTION::create);
+
         try {
-            // step 2 encrypt the encrypted message using the public key of the recipient
-            PlayerPublicKey ppk = entry.get().getPublicKeyData();
+            PlayerPublicKey ppk = recipient.getPublicKeyData();
             if (ppk == null) {
-                source.sendError(Text.translatable("commands.cwhisperencrypted.pubKeyNotFound"));
-                return 0;
+                throw PUBLIC_KEY_NOT_FOUND_EXCEPTION.create();
             }
             PublicKey publicKey = ppk.data().key();
             Cipher cipher = Cipher.getInstance("RSA/ECB/PKCS1Padding");
             cipher.init(Cipher.ENCRYPT_MODE, publicKey);
-            byte[] encrypted = message.getBytes(StandardCharsets.UTF_8);
-            encrypted = Gzip.compress(encrypted);
-            if (encrypted.length > 245) {
-                source.sendError(Text.translatable("commands.cwhisperencrypted.messageTooLong"));
-                return 0;
+            byte[] bytes = message.getBytes(StandardCharsets.UTF_8);
+            byte[] compressedBytes = Gzip.compress(bytes);
+            if (compressedBytes.length > 245) {
+                throw MESSAGE_TOO_LONG_EXCEPTION.create();
             }
-            cipher.update(encrypted);
-            encrypted = cipher.doFinal();
-            String commandMessage = "w " + entry.get().getProfile().getName() + " CCENC:" + BaseUTF8.toUnicode(encrypted);
+            cipher.update(compressedBytes);
+            compressedBytes = cipher.doFinal();
+            String commandMessage = "w " + recipient.getProfile().getName() + " CCENC:" + BaseUTF8.toUnicode(compressedBytes);
             if (commandMessage.length() >= 256) {
-                source.sendError(Text.translatable("commands.cwhisperencrypted.messageTooLong"));
-                return 0;
+                throw MESSAGE_TOO_LONG_EXCEPTION.create();
             }
+            Text preview = Util.map(source.getClient().inGameHud.getChatHud().getChatScreen().getChatPreviewer().tryConsumeResponse("/" + commandMessage), ChatPreviewer.Response::previewText);
+            source.getPlayer().sendCommand(commandMessage, preview);
```

Comment:
You dont need to create the preview iirc. Its `@nullable`

Reply:
Right. It used to be omitted though, which didn't work anymore.

Reply:
So passing null doesnt work anymore?

Reply:
No not passing `null`, omitting the argument. I am currently testing passing `null`.

Reply:
Passing `null` seems to work, I will change it.